tests/CMakeLists.txt

```diff
@@ -1 +1,50 @@
+# Add an executable and its related test.
+#
+# The executable is always linked to 'kokkostools' and 'test_common'.
+#
+# Arguments:
+#   TARGET_NAME : name of the test
+#   SOURCE_FILE : source file, defaults to <TARGET_NAME>.cpp
+function(kp_add_executable_and_test)
+
+    cmake_parse_arguments(kaeat_args "" "TARGET_NAME;SOURCE_FILE" "" ${ARGN})
+
+    if(NOT DEFINED kaeat_args_TARGET_NAME)
+        message(FATAL_ERROR "Wrong usage.")
+    endif()
+
+    if(NOT DEFINED kaeat_args_SOURCE_FILE)
+        set(kaeat_args_SOURCE_FILE "${kaeat_args_NAME}.cpp")
+    endif()
```

Comment:
This isn't used and it's not clear to me that setting a default source file improves behavior (instead of just erroring out).